src/plugins/discover/public/utils/get_should_show_field_handler.test.ts

```diff
@@ -74,20 +74,24 @@ describe('get fields to show', () => {
   });
 
   it('shows multifields when showMultiFields is true', () => {
-    const fieldsToShow = getFieldsToShow(
+    const shouldShowFieldHandler = getShouldShowFieldHandler(
       ['machine.os', 'machine.os.raw', 'clientip'],
       dataView,
       true
     );
-    expect(fieldsToShow).toEqual(['machine.os', 'machine.os.raw', 'clientip']);
+    expect(shouldShowFieldHandler('machine.os')).toBe(true);
+    expect(shouldShowFieldHandler('machine.os.raw')).toBe(true);
+    expect(shouldShowFieldHandler('clientip')).toBe(true);
   });
 
   it('do not show multifields when showMultiFields is false', () => {
-    const fieldsToShow = getFieldsToShow(
+    const shouldShowFieldHandler = getShouldShowFieldHandler(
       ['machine.os', 'machine.os.raw', 'acknowledged', 'clientip'],
       dataView,
       false
     );
-    expect(fieldsToShow).toEqual(['machine.os', 'acknowledged', 'clientip']);
+    expect(shouldShowFieldHandler('acknowledged')).toBe(true);
+    expect(shouldShowFieldHandler('clientip')).toBe(true);
+    expect(shouldShowFieldHandler('machine.os.raw')).toBe(false);
```

Comment:
Maybe we should add  the check for `machine.os` so it's the same check like before?